### PR TITLE
initialise MPS with certain data type T<:Number

### PR DESCRIPTION
**Pull request by jtschneider**
Perhaps a `function MPS(::Type{T},sites) where {T<:Number}` might be appreciated in case of initialising an MPS with e.g. `ComplexF64` data type.

### CODE REVIEW

**Comment by mtfishman** [comment]
Definitely useful, thanks. What if we define `MPS(sites) = MPS(Float64,sites)` to avoid code duplication?

**Comment by jtschneider** [comment]
Sounds quite reasonable. Will do.

**Comment by orialb** [comment]
It will be also very useful to have this feature in `productMPS` which I guess is one of the most common ways for users to initialize an `MPS` for physics applications.

File: src/mps/mps.jl [hunk 1]
```diff
@@ -26,30 +26,32 @@ mutable struct MPS
...
-      v[ii] = ITensor(l[ii], s)
-    elseif ii == N
-      v[ii] = ITensor(l[ii-1], s)
-    else
-      v[ii] = ITensor(l[ii-1],s,l[ii])
-    end
-  end
-  return MPS(N,v,0,N+1)
-end
+function MPS(::Type{T},sites) where {T<:Number}
```
Comment (mtfishman) [inline src/mps/mps.jl]: The formatting of this function looks off, could you please clean it up? Thanks!

Comment (jtschneider) [reply src/mps/mps.jl]: Yes, don't know what happened there...

**Comment by mtfishman** [comment]
Looks good, thanks!

**Pull request merged**